src/observers/http2/http2_observer.c

```diff
@@ -246,7 +240,33 @@ static int on_header_recv_callback(qd_http2_decoder_connection_t *conn_state,
             }
         }
     } else if (strcmp(X_FORWARDED_FOR, (const char *)name) == 0) {
-        set_stream_vflow_attribute(transport_handle, stream_id, VFLOW_ATTRIBUTE_SOURCE_HOST, (const char *)value);
+        qd_error_t error = get_stream_info_from_hashtable(transport_handle, &stream_info, stream_id);
+        if (error == QD_ERROR_NOT_FOUND) {
+            qd_log(LOG_HTTP2_DECODER, QD_LOG_ERROR, "[C%"PRIu64"] set_stream_vflow_attribute could not find in the hashtable, stream_id=%" PRIu32, transport_handle->conn_id, stream_id);
+        }
+        else {
+            if (!stream_info->x_forwarded_for) {
+                //
+                // We will capture the very first x-forwarded-for header and ignore the other x-forwarded for headers in the same request.
+                // Say a request has the following three x-forwarded-for headers
+                //
+                // X-Forwarded-For: 2001:db8:85a3:8d3:1319:8a2e:370:7348, 197.1.773.201
+                // X-Forwarded-For: 195.0.223.001
+                // X-Forwarded-For: 203.0.113.195, 2007:db5:85a3:8d3:1319:8a2e:370:3221
+                //
+                // We will obtain the value of the first X-Forwarded-For (2001:db8:85a3:8d3:1319:8a2e:370:7348, 197.1.773.201) and ignore the
+                // other two X-Forwarded-For headers.
+                // The first X-Forwarded-For header is comma separated list, we will obtain the leftmost (the first) value (2001:db8:85a3:8d3:1319:8a2e:370:7348) in the list
+
+                // const uint8_t *value passed into this function is guaranteed to be NULL-terminated
+                char value_copy[valuelen+1];
```

Comment:
Is there a limit to valuelen?  Can it be large enough to overflow the stack?

Reply:
> so hardcoding a array size of 128 would be more than enough I would think.

Agreed. I made the change. Thanks @kgiusti